### ISSUE TITLE
Problems panel lists 4 problems for jsconfig.json in my project

### ISSUE THREAD

**Issue by lewisgauss** [microsoft/vscode#116528]
- VS Code Version: 1.53.2
- OS Version: Windows 10

Steps to Reproduce:

1. Create a new Gatsby project, or clone from [my repo](https://github.com/lewisgauss/lewischan.dev)
2. Open VSCode
3. Create a new jsconfig.json file if it does not exist
4. Problems panel will randomly show errors on the jsconfig.json file
![image](https://user-images.githubusercontent.com/8966669/107747793-fcbe2e00-6d52-11eb-93d9-f3cc08935d89.png)

The jsconfig.json file contains this:
```json
{
    "compilerOptions": {
        "module": "commonjs",
        "jsx": "preserve",
        "baseUrl": "./src"
    },
    "include": ["src"],
    "exclude": [
        "node_modules",
        "**/node_modules/*",
        "node_modules/**/*",
        "public"
    ]
}
```

Does this issue occur when all extensions are disabled?: Yes

**Comment by gjsjohnmurray** [comment]
/needsMoreInfo

Please run the Extension Bisect command to determine whether these are coming from an extension you added (you didn't list these in your report). If Bisect identifies an extension, report the issue to the extension author. The "Report Issue" dialog on the "Help" menu can assist with that. Then close this issue, and optionally link it to the one you created on the extension's repository.

**Comment by lewisgauss** [comment]
Hi @gjsjohnmurray 

I just ran it with Extension Bisect, and the issue is still there. 

![image](https://user-images.githubusercontent.com/8966669/107760349-8d513a00-6d64-11eb-936e-ba94f5c40b4b.png)

**Comment by mjbvz** [comment]
Please provide a minimal sample project and specific steps to reproduce the problem

**Comment by lewisgauss** [comment]
Hi @mjbvz 

You can use the project I in first comment. just create a new .js file inside src folder, that is already good enough to trigger the problem to appear. Or do you need something more simple than that?

In the repo I shared, you can use the branch "reboot" to reproduce the problem. 

1. Run `yarn install` after checkout to reboot branch
2. If VSCode is already open, restart it
3. Create a new folder in components
4. Create a new file, with the same folder name

Then the problems will pop up. If creating 1 file does not trigger it, you can create more files in the folder you have created, or create new folders and new files to trigger it.

**Comment by mjbvz** [comment]
Please provide a minimal project. The project you linked to is quite large and this type of bug is often a project configuration issue. It's not a good use of our time to try debugging through it

**Comment by lewisgauss** [comment]
Thanks the for reply, @mjbvz 

I will try to provide one. 

I actually specifically mentioned the reboot branch, because that branch is new, and it only has the bare minimum Gatsby startup code, with Airbnb's ESLint packages. 

If you look at the main vs reboot, you will find that reboot is much smaller.

**Comment by lewisgauss** [comment]
Hi @mjbvz 

I have been able to reliably reproduce the problem using this new minimal project - [lewisgauss/reproduce-jsconfig-error](https://github.com/lewisgauss/reproduce-jsconfig-error)

I also have a much simpler way to reproduce it. 

1. Open the project
2. Open any .js file, like pages/index.js 
3. Open jsconfig.json
4. In jsconfig.json, edit it you can remove one line, like jsx: "preserve"
5. Click on the .js file that you opened, so Visual Studio Code can reinitialize the feature
6. Click on jsconfig.json again, add the line that you just removed back
7. Click on the .js file again, the problems will now show in the panel

**Comment by mjbvz** [comment]
I followed those steps but this does not reproduce for me in VS Code insiders with extensions disabled

**Comment by lewisgauss** [comment]
@mjbvz did you do npm install before going through the steps?

You also have to wait for the initializing JS/TS feature to finish in between steps. 
![image](https://user-images.githubusercontent.com/8966669/108166768-9dd02e80-712f-11eb-8698-20ec9537d631.png)

I have just tried this on a completely different computer, and I could reproduce it. (macOS and Windows 10) With all extensions disabled. :) 

![image](https://user-images.githubusercontent.com/8966669/108166712-8a24c800-712f-11eb-8678-7ae385ef4153.png)

**Comment by lewisgauss** [comment]
Another run with Extension Bisect. 

![image](https://user-images.githubusercontent.com/8966669/108167633-f05e1a80-7130-11eb-9f0f-e78179f77e6e.png)

**Comment by mjbvz** [comment]
Yes node modules are installed and initialization is done.

1. Does `node_modules\defer-to-connect\dist\index.d.ts` exist and are there any errors in it?
2. If it does, then please record a video showing the exact steps you are taking here so I can try to match them

**Comment by lewisgauss** [comment]
Yes, the file is there, and TSLint does not show any error. You can see it from the recording below, which shows the reproducible steps and the file without error. 

https://user-images.githubusercontent.com/8966669/108243721-5c6c6d00-7189-11eb-8303-993b96864aad.mp4

**Comment by mjbvz** [comment]
Thanks but I am still not able to reproduce following those steps. Can you try collecting the logs from reproducing this:

1. Set `"typescript.tsserver.log": "verbose",`
1. Restart VS Code and reproduce the problem
1. In VS Code, run the `TypeScript: Open TS Server log` command
1. This should open a large log file called `tsserver.log` 

 > **⚠️Warning:** The TypeScript log may include information from your workspace, including file paths and source code. If you have any concerns about posting this publicly on Github, just let me know and we can arrange something else. On our side, we only use these logs to investigate issues like this

**Comment by lewisgauss** [comment]
Hello again, @mjbvz 

The file is available [here](https://gist.github.com/lewisgauss/debff1fdf6dc6d87141c8dbe92501f92). Hope this helps!
